Credit-Conveyer/Credit-Conveyer/src/test/java/com/enfint/CreditConveyer/controller/OfferAndCalculationControllerTest.java

```diff
@@ -0,0 +1,116 @@
+package com.enfint.CreditConveyer.controller;
+
+import com.enfint.CreditConveyer.dto.EmploymentDTO;
+import com.enfint.CreditConveyer.dto.LoanApplicationRequestDTO;
+import com.enfint.CreditConveyer.dto.LoanOfferDTO;
+import com.enfint.CreditConveyer.dto.modelenum.EmploymentStatus;
+import com.enfint.CreditConveyer.dto.modelenum.Gender;
+import com.enfint.CreditConveyer.dto.modelenum.MaritalStatus;
+import com.enfint.CreditConveyer.dto.modelenum.Position;
+import com.enfint.CreditConveyer.dto.ScoringDataDTO;
+import com.enfint.CreditConveyer.service.OfferAndCalculationService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+@WebMvcTest(controllers = OfferAndCalculationController.class)
+class OfferAndCalculationControllerTest {
+
+    /*With the help of SpringJUnit4ClassRunner and MockMVc*/
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper mapper= new ObjectMapper();
+
+    @Autowired
+    private JsonMapper jsonMapper= new JsonMapper();
+
```

Comment:
@Autowired is not needed here
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.enfint.CreditConveyer.controller.OfferAndCalculationControllerTest': Unsatisfied dependency expressed through field 'jsonMapper';